src/components/MemoMainPage.js

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useState } from "react";
+import MemoList from "./MemoList";
+import AddNewMemoButton from "./AddNewMemoButton";
+import MemoForm from "./MemoForm";
+
+export default function MemoMainPage({ action, setAction }) {
+  const initialMemos = () => {
+    try {
+      const allMemosJson = localStorage.getItem("memos");
+      const initialMemos =
+        // If not exist "memos" key, return value is null.
+        allMemosJson === null ? [] : JSON.parse(allMemosJson);
+      return initialMemos;
+    } catch (err) {
+      // Private browser may raise an exception.
+      if (err instanceof Error) {
+        console.error(err);
+      } else {
+        throw err;
+      }
+    }
+  };
+
+  const [allMemos, setAllMemos] = useState(initialMemos);
+  const [formText, setFormText] = useState("");
+  const [editingMemo, setEditingMemo] = useState(null);
+
+  useEffect(() => {
+    try {
+      localStorage.setItem("memos", JSON.stringify(allMemos));
+    } catch (error) {
+      // If storage capacity is full, setItem raises an exception.
+      if (error instanceof Error) {
+        console.error(error.message);
+      } else {
+        throw error;
+      }
+    }
+  }, [allMemos]);
+
+  // Event Handlers
+  const handleAddNewMemoButtonClick = () => {
+    const newMemo = { id: crypto.randomUUID(), content: "新規メモ" };
+
+    const nextMemos = [...allMemos, newMemo];
+    setAllMemos(nextMemos);
+    openMemoForm(newMemo, "new");
+  };
+
+  const handleEditButtonClick = () => {
+    if (formText === null || formText === undefined || formText.trim() === "") {
+      alert("保存するメモの内容を書いてください。");
+      return;
+    }
+
+    const updatedMemos = allMemos.map((memo) =>
+      memo.id === editingMemo.id ? { ...memo, content: formText } : memo,
+    );
+
+    setAllMemos(updatedMemos);
+    resetPage("index");
+  };
+
+  const handleDeleteButtonClick = () => {
+    const oneLessMemos = allMemos.filter((memo) => memo.id !== editingMemo.id);
+
+    setAllMemos(oneLessMemos);
+    resetPage("index");
+  };
+
+  const handleMemoTitleClick = (clickedMemo) => {
+    const memo = allMemos.find((memo) => clickedMemo.id === memo.id);
```

Comment:
検索の主体は `memo` なので、オペランドを入れ替えたほうがよさそうです。

```suggestion
    const memo = allMemos.find((memo) => memo.id === clickedMemo.id);
```

現状のコードだと、以下のような式になっています。

```js
if (0 === array.length) {
  ...
}
```